Address issue #70 and create an error message if old player.json is used

joshurtree opened the pull request:
Converts the 'Show unowned crew' checkbox to a dropdown with the options of "Show all", "Only show unowned crew" and "Only show owned crew". Also introduces an error message when the user uses a player.json file with client_id < 17 instead of a blank page.

ussjohnjay commented:
That's a nice addition and it works well! Just a few suggestions:

1) Could we match the style of the minimum rarity field (or vice versa)? They look like different types of elements, even though I think they should be the same.

2) Could we filter out immortals from the owned list, so that it's more like "Owned crew I need stars for"?

3) Could we only show the new error after the fetching of keystones and crew is completed and the data is found to be invalid? Right now it shows the error while the fetching is occurring, which is somewhat misleading, even if it's only for a second or so.

ineffyble commented:
Huh, it hadn't occurred to me that people would try with old versions, but makes a lot of sense. Thanks!

ineffyble commented on src/components/crewretrieval.tsx:
```diff
@@ -186,17 +206,18 @@ class CrewRetrieval extends Component<CrewRetrievalProps, CrewRetrievalState> {
-								checked={this.state.unownedOnly}
-								onChange={(e, { checked }) => this.setState({ unownedOnly: checked })}
+								control={Dropdown}
+								inline
```
```suggestion
								inline
								selection
```

This should fix the styling discrepancy mentioned by @ussjohnjay.

ineffyble commented:
> 2. Could we filter out immortals from the owned list, so that it's more like "Owned crew I need stars for"?

Agreed but we need to make it explicit. Perhaps the option should be "Only show unowned crew not FF" or such?

To implement that we'll need to find the max stars on a copy of a crew in player inventory. Once we have done that it should be trivial to implement #60 by passing the player_roster_max_stars_for_crew (not suggesting this should be final name) to the Rating component where we currently pass `crew.rarity`. So if you want to tackle that too, that would be amazing.

joshurtree commented:
I've addressed points 1 and 2 

> Huh, it hadn't occurred to me that people would try with old versions, but makes a lot of sense. Thanks!

In my experience IT project always suffer from the problem that their designers/programmers only consider how their product should be used and not how it might be misused. Every project needs an idiot that prods at the corners that the designers didn't think go to. I can be that idiot.

ineffyble commented on src/components/crewretrieval.tsx:
```diff
@@ -161,16 +179,20 @@ class CrewRetrieval extends Component<CrewRetrievalProps, CrewRetrievalState> {
-		const { column, direction, pagination_rows, pagination_page, unownedOnly, minRarity } = this.state;
+		const { column, direction, pagination_rows, pagination_page, ownedFilter, minRarity } = this.state;
 		let { data } = this.state;
 		if (!data) {
```
```suggestion
                 const { playerData } this.props;
		if (!data) {
		  if (playerData.forte_root) {
		    // crew/keystones are loading, return null while we wait (could be replaced with a loading icon)
		    return null;
		  } else {
		   // no keystone data available, file must be too old
```

ineffyble reviewed (approved):
Looks great, thanks!
